### ISSUE TITLE
Cannot add new services (previous saved data is lost)

### ISSUE THREAD

**Issue by ayxos** [getferdi/ferdi#1945]
**Bug description:**
After Snap update, All previous data is lost and I cannot add services

**Steps to reproduce:**

1. Go to 'Get Started'
2. Click on a service
3. Nothing happens

**Expected behavior**
selected service being added and ready to configure

**Environment:**

Version: 5.6.2
Electron: 13.3.0
Chrome: 91.0.4472.164
Node.js: 14.16.0
Platform: Linux 5.11
Arch: x64
Build date: Mon Sep 13 2021 10:17:08 GMT+0200 (Central European Summer Time)
Git SHA: 40a3d0b
Git branch: release

**Comment by rplevka** [comment]
I confirm.

developer tools console lists several errors:

```
Error: EPERM: operation not permitted, copyfile '/home/rplevka/snap/ferdi/48/.config/Ferdi/recipes/temp/hangoutschat/icon.svg' -> '/home/rplevka/snap/ferdi/48/.config/Ferdi/recipes/hangoutschat/icon.svg'
    at Object.copyFileSync (fs.js:1991)
    at Object.func [as copyFileSync] (electron/js2c/asar_bundle.js:5)
    at copyFile (node_modules/fs-extra/lib/copy-sync/copy-sync.js:70)
    at onFile (node_modules/fs-extra/lib/copy-sync/copy-sync.js:56)
    at getStats (node_modules/fs-extra/lib/copy-sync/copy-sync.js:48)
    at startCopy (node_modules/fs-extra/lib/copy-sync/copy-sync.js:38)
    at copyDirItem (node_modules/fs-extra/lib/copy-sync/copy-sync.js:122)
    at node_modules/fs-extra/lib/copy-sync/copy-sync.js:115
    at Array.forEach (<anonymous>)
    at copyDir (node_modules/fs-extra/lib/copy-sync/copy-sync.js:115)
    at onDir (node_modules/fs-extra/lib/copy-sync/copy-sync.js:105)
    at getStats (node_modules/fs-extra/lib/copy-sync/copy-sync.js:45)
    at handleFilterAndCopy (node_modules/fs-extra/lib/copy-sync/copy-sync.js:33)
    at copySync (node_modules/fs-extra/lib/copy-sync/copy-sync.js:26)
    at ServerApi.getRecipePackage (api/server/ServerApi.js:1)
    at async api/server/ServerApi.js:1
    at async Promise.all (index 2)
    at async ServerApi._mapServiceModels (api/server/ServerApi.js:1)
    at async ServerApi.getServices (api/server/ServerApi.js:1)
(anonymous) @ instrument.ts:129
```

```
Can't load recipe Error: Cannot find module 'airtable'
Require stack:
- /snap/ferdi/48/resources/app.asar/api/server/ServerApi.js
- /snap/ferdi/48/resources/app.asar/index.html
    at Module._resolveFilename (internal/modules/cjs/loader.js:887)
    at Function.o._resolveFilename (electron/js2c/renderer_init.js:33)
    at Module._load (internal/modules/cjs/loader.js:732)
    at Function.f._load (electron/js2c/asar_bundle.js:5)
    at Function.o._load (electron/js2c/renderer_init.js:33)
    at Module.require (internal/modules/cjs/loader.js:959)
    at require (internal/modules/cjs/helpers.js:88)
    at api/server/ServerApi.js:1
    at Array.map (<anonymous>)
    at ServerApi.getInstalledRecipes (api/server/ServerApi.js:1)
    at api/server/ServerApi.js:1
    at async Promise.all (index 2)
    at async ServerApi._mapServiceModels (api/server/ServerApi.js:1)
    at async ServerApi.getServices (api/server/ServerApi.js:1)
(anonymous) @ instrument.ts:129
```

**Comment by vraravam** [comment]
I dont use linux (and over 10 years ago when I was on ubuntu, I didnt use snap if it was even there), but, based on other conversations with @mhatvan , isn't that how snaps are supposed to work? ie each upgrade creates a new profile folder?

**Comment by rplevka** [comment]
There are some log messages about 45->48 migrations, so i guess there's some mechanism that deals with this

**Comment by mhatvan** [comment]
I use `.deb` instead of `snap` since I need to be able to switch ferdi version a lot as maintainer, so haven't experienced these issues myself.

Since `snap` is a self-contained sandbox, I am wondering how it preserves settings from one version to the next, I guess by copying the whole folder to the next version.

While `.deb` Ferdi uses `~/.config/Ferdi` location which persists across version upgrades as well as Ferdi deletion, `snap` uses `~/snap/ferdi/<snap-version>/.config/Ferdi` which seems to be problematic.

I wonder if it is possible to point snap to a custom location on the users file system. Probably not.

**Comment by EpoxyD** [comment]
Checked using the .deb installation: removing the ~/.config/Ferdi folder also fixes the issue, although you have to setup Ferdi again.

**Comment by vraravam** [comment]
@EpoxyD - that is why my recommendation was to only remove the `recipes` folder inside the profile folder (`/.config/Ferdi` as you mentioned in your OS)

**Comment by redhottaco** [comment]
I'm having the same issue but on win10.

**Comment by vraravam** [comment]
Which server are you using Ferdi with? There's a known issue if you are using Ferdi with the externally hosted Ferdi server.

**Comment by redhottaco** [comment]
Ah, I am using the Ferdi server.

**Comment by vraravam** [comment]
That explains it. Closing this issue, since this is being tracked [here](https://github.com/getferdi/server/issues/74)

**Comment by carbeer** [comment]
I installed Ferdi just a few minutes ago through snap. I'm using default settings (which I assume is the internal server and not the external hosted server - I didn't set anything up) and face the same issue, I was unable to add any service with or without account. 

Removing `~/snap/ferdi/` didn't help. I also noticed that after trying to add a service, files with the service name appear in the `~/snap/ferdi/48/.config/Ferdi/recipes/temp` directory. 

If snap at this time is discouraged as installation method? It is still listed on the homepage.
